test/e2e/page-objects/tribunals-case-api/getCaseData.js

```diff
@@ -0,0 +1,41 @@
+const config = require('config');
+
+const tribunalsApiUrl = config.get('api.url');
+const authCookie = '__auth-token';
+
+async function getMYACaseData(I, ccdCaseID) {
+  const res = await I.sendGetRequest(`${tribunalsApiUrl}/appeals?caseId=${ccdCaseID}`);
+
+  if (res && res.status === 200) {
+    if (res.data && res.data.appeal) {
+      return res.data.appeal;
+    }
+    throw Error('Invalid API Response)');
+  } else {
+    const status = res && res.status ? res.status : 'null';
+    throw Error(`HTTP Error ${status} not correct`);
+  }
+}
+
+async function getCaseData(I, ccdCaseID) {
+  const myaCaseData = await I.getMYACaseData(I, ccdCaseID);
+  if (!myaCaseData || !myaCaseData.appealNumber) {
+    throw Error('Invalid Appeal Number)');
+  }
+  const tyaID = myaCaseData.appealNumber;
+  const authTokenCookie = await I.grabCookie(authCookie);
+  const headers = { Authorization: `Bearer ${authTokenCookie.value}` };
+  const res = await I.sendGetRequest(`${tribunalsApiUrl}/api/citizen/${tyaID}`, headers);
+
+  if (res && res.status === 200) {
+    if (res.data) {
+      return res.data;
+    }
+    throw Error('Invalid API Response)');
```

Comment:
Same as above.

You could actually extract this whole block into a common function and pass in different functions for what to return. The only difference is:

if (res.data && res.data.appeal) {
      return res.data.appeal;
    }
    
and

if (res.data) {
      return res.data;
    }

Reply:
This would mean you can cut the number of tests in ~half too